tests/mod_test.py

```diff
@@ -0,0 +1,75 @@
+from .context import assert_equal
+import pytest
+from sympy import Symbol, Rational, Mod, sqrt, nsimplify, pi, GoldenRatio
+from sympy.physics.units import hbar
+
+x = Symbol('x', real=True)
+y = Symbol('y', real=True)
+
+
+def test_mod_usual():
+    assert_equal("128\\mod 3", Mod(128, 3))
+    assert_equal("7\\mod 128", Mod(7, 128))
+    assert_equal("5\\mod 10", Mod(5, 10))
+    assert_equal("5\\mod 5", Mod(5, 5))
+    assert_equal("3\\mod 2", Mod(3, 2))
+    assert_equal("0 \\mod 6", Mod(0, 6))
+    assert_equal("6109\\mod 28", Mod(6109, 28))
+    assert_equal("4000000000\\mod 28791", Mod(4000000000, 28791))
+    assert_equal("128*10^300\\mod 876123", Mod(Rational('128E300'), 876123))
+    assert_equal("876,123\\mod 128E300)", Mod(876123, Rational('128E300')))
+
+
+def test_mod_negative():
+    assert_equal("-1\\mod 2", Mod(-1, 2))
+    assert_equal("-3\\mod 3", Mod(-3, 3))
+    assert_equal("-12\\mod -12", Mod(-12, -12))
+    assert_equal("-128\\mod 4", Mod(-128, 4))
+    assert_equal("9\\mod -213", Mod(9, -213))
+    assert_equal("123123\\mod -541", Mod(123123, -541))
+    assert_equal("-123123\\mod 541", Mod(-123123, 541))
+    assert_equal("-97E34\\mod 7", Mod(Rational('-97E34'), 7))
+
+
+def test_mod_fraction():
+    assert_equal("1/2\\mod 3", Mod(Rational(1, 2), 3))
+    assert_equal("6/2\\mod 3", Mod(Rational(6, 2), 3))
+    assert_equal("-14/2\\mod 5", Mod(Rational(-14, 2), 5))
+    assert_equal("123\\mod (42/6)", Mod(123, Rational(42, 6)))
+    assert_equal("431\\mod (2/123)", Mod(431, Rational(2, 123)))
+    assert_equal("5/5\\mod (5/5)", Mod(Rational(5, 5), Rational(5, 5)))
+    assert_equal("849/-21\\mod (092/2)", Mod(Rational(849, -21), Rational(92, 2)))
+    assert_equal("13*10^9\\mod (21/-2)", Mod(13E9, Rational(21, -2)))
+
+
+def test_mod_float():
+    assert_equal("0.41\\mod 2", Mod(Rational('0.41'), 2))
+    assert_equal("143E-13\\mod 21", Mod(Rational('143E-13'), 21))
+    assert_equal("-9.80665\\mod 9.80665", Mod(-9.80665, 9.80665))
+    assert_equal("0.0000923423\\mod -8341.234802909", nsimplify(Mod(0.0000923423, -8341.234802909)))
+    assert_equal("\\sqrt{5}\\mod \\sqrt{2}", Mod(sqrt(5), sqrt(2)))
+    assert_equal("987\\mod \\pi", Mod(987, pi))
+    assert_equal("\\pi\\mod ((1+\\sqrt{5})/2) ", Mod(pi, GoldenRatio))
```

Comment:
@tirkarp now that I'm running pytest, I'm getting a failure on this one. 
`actual = -sqrt(5)/2 - 1/2 + pi, expected = pi - GoldenRatio`

Reply:
That should've been caught on my end, but it passed on my machine. Hopefully it's fixed now with the new commit.